test/test_nestedtensor.py

```diff
@@ -355,6 +355,54 @@ def test_device_checks(self, device):
         is_cuda = 'cuda' in str(device)
         self.assertEqual(nt.is_cuda, is_cuda)
 
+    @dtypes(torch.float, torch.float16, torch.double)
+    def test_nested_tensor_indexing(self, device, dtype):
+        # edge case: empty nested tensor
+        nt0 = torch.nested_tensor([])
+        self.assertRaisesRegex(
+            RuntimeError,
+            "cannot index an empty nested tensor",
+            lambda: nt0[0]
+        )
+        # normal case
+        x0 = torch.randn((2, 5), device=device)
+        x1 = torch.randn((3, 4), device=device)
+        nt = torch.nested_tensor([x0, x1])
+        # single index: only support integer in the batch dimension
+        self.assertEqual(nt[0], x0)
+        self.assertEqual(nt[-1], x1)
+        self.assertRaisesRegex(
+            IndexError,
+            "index 2 is out of bounds for nested tensor dimension 0 with size 2",
+            lambda: nt[2]
+        )
+        self.assertRaisesRegex(
+            IndexError,
+            "index -3 is out of bounds for nested tensor dimension 0 with size 2",
+            lambda: nt[-3]
+        )
+        self.assertRaises(NotImplementedError, lambda: nt[:])
+        self.assertRaises(NotImplementedError, lambda: nt[None])
+        self.assertRaises(NotImplementedError, lambda: nt[...])
+        # tuple of indices: only support integer in the batch dimension
+        #                 + all possible indexing in the original tensor dimensions
+        self.assertEqual(nt[0, 0, 0], x0[0, 0])
+        self.assertEqual(nt[0, 1, :], x0[1, :])
+        self.assertEqual(nt[1, ...], x1)
+        self.assertRaisesRegex(
+            IndexError,
+            "index 4 is out of bounds for dimension 1 with size 3",
```

Comment:
nit: Technically this error message is incorrect, since dimension 1 of this particular NestedTensor doesn't have a size (it's irregular). 

```
x0 = torch.randn((2, 5), device=device)
x1 = torch.randn((3, 4), device=device)
nt = torch.nested_tensor([x0, x1])
```
has nested size
```
[ 
  (2, 5),
  (3, 4)
]
```

Reply:
Yeah that's an issue. In 2nd and latter `handleDimInMultiDimIndexing` calls, it works on the original normal tensor, which means the error messages become the ones for the original normal tensor. I can't think of a way to get rid of it

Reply:
Makes sense. I'd note that as a follow-up item and create a small, separate issue to track it.